dump/plugins/vlan.py

```diff
@@ -0,0 +1,157 @@
+                self.ret_temp[req.db]["keys"].append(mem)
+        else:
+            self.ret_temp[req.db]["tables_not_found"].append(req.table)
+
+    def init_asic_vlan_info(self, vlan_name):
+        # Convert 'Vlanxxx' to 'xxx'
+        if vlan_name[0:4] != "Vlan" or not vlan_name[4:].isnumeric():
+            self.ret_temp["ASIC_DB"]["tables_not_found"] =["ASIC_STATE:SAI_OBJECT_TYPE_VLAN"]
+            return {}, {}
+        vlan_num = int(vlan_name[4:])
+        
+        # Find the table named "ASIC_STATE:SAI_OBJECT_TYPE_VLAN:*" in which SAI_VLAN_ATTR_VLAN_ID = vlan_num
+        req = MatchRequest(db="ASIC_DB", table="ASIC_STATE:SAI_OBJECT_TYPE_VLAN", key_pattern="*", field="SAI_VLAN_ATTR_VLAN_ID", 
+                           value=str(vlan_num), ns=self.ns)
+        ret = self.match_engine.fetch(req)
+        if not ret["error"] and len(ret["keys"]) != 0:
+            self.ret_temp[req.db]["keys"] = ret["keys"]
+        else:
+            self.ret_temp[req.db]["tables_not_found"] = [req.table]
+            
+        # Return request and table to caller in case its next call is to init_asic_vlan_member_info which also needs it
+        return req, ret
+    
+    def init_asic_vlan_member_info(self, vlan_name, req, vlan_table):
+        
+        bridge_oids = []
+        ret = {}
+        
+        # Handle invalid vlan name 
+        if vlan_name[0:4] != "Vlan" or not vlan_name[4:].isnumeric():
+            self.ret_temp["ASIC_DB"]["tables_not_found"].append("ASIC_STATE:SAI_OBJECT_TYPE_VLAN_MEMBER")
+            self.ret_temp["ASIC_DB"]["tables_not_found"].append("ASIC_STATE:SAI_OBJECT_TYPE_BRIDGE_PORT")
+            return
+        
+        # Find vlan table if not passed by caller
+        if not req or not vlan_table:
+        
+            # Convert 'Vlanxxx' to 'xxx'
+            vlan_num = int(vlan_name[4:])
+    
+            # Find the table named "ASIC_STATE:SAI_OBJECT_TYPE_VLAN:*" in which SAI_VLAN_ATTR_VLAN_ID = vlan_num
+            req = MatchRequest(db="ASIC_DB", table="ASIC_STATE:SAI_OBJECT_TYPE_VLAN", key_pattern="*", 
+                               field="SAI_VLAN_ATTR_VLAN_ID", value=str(vlan_num), ns=self.ns)
+            vlan_table = self.match_engine.fetch(req)
+
+        # Find all the member tables whose vlan is the oid:0x... part of the table name just found 
+        if not vlan_table["error"] and len(vlan_table["keys"]) != 0:
+            req = MatchRequest(db="ASIC_DB", table="ASIC_STATE:SAI_OBJECT_TYPE_VLAN_MEMBER", key_pattern="*", 
+                               field="SAI_VLAN_MEMBER_ATTR_VLAN_ID", value=vlan_table["keys"][0][32:], 
+                               return_fields=["SAI_VLAN_MEMBER_ATTR_BRIDGE_PORT_ID"], ns=self.ns)
+            ret = self.match_engine.fetch(req)
+
+        # Retrieve SAI_OBJECT_TYPE_VLAN_MEMBER and SAI_OBJECT_TYPE_BRIDGE_PORT tables for each member port 
+        if ret and not ret["error"] and len(ret["keys"]) != 0:
+            for mem in ret["keys"]:
```

Comment:
In this logic if there are many members of vlan, then each db will contain a list of tables with no correlation between them and it would defeat the purpose of dump infra. Please separate out vlan and vlan_member as separate modules.